### ISSUE TITLE
Transfer state does not escape double quotes

### ISSUE THREAD

**Issue by kgajera** [scullyio/scully#1186]
### Description

If an API returns JSON that contains double quotes:

```
{
  "test": "This sentence has a \" double quote in it"
}
```
An error occurs when `JSON.parse` is called on the saved state:

```
Uncaught SyntaxError: Unexpected token d in JSON at position 44
```

#### 🔬 Minimal Reproduction

https://github.com/kgajera/scully-transfer-state-bug
https://github.com/kgajera/scully-transfer-state-bug/blob/main/src/app/app.component.ts

#### 💻Your Environment

**Angular Version:**

<pre><code>
Angular CLI: 11.0.5
Node: 14.15.2
OS: darwin x64

Angular: 11.0.5
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1002.0
@angular-devkit/build-angular   0.1100.5
@angular-devkit/core            11.0.5
@angular-devkit/schematics      11.0.5
@angular/cdk                    11.0.3
@angular/fire                   6.1.4
@angular/flex-layout            11.0.0-beta.33
@angular/material               11.0.3
@schematics/angular             11.0.5
@schematics/update              0.1100.5
rxjs                            6.6.3
typescript 
</code></pre>

**Scully Version:**

<pre><code>
"@scullyio/init": "^1.0.1",
"@scullyio/ng-lib": "^1.0.0",
"@scullyio/scully": "^1.0.0",
</code></pre>

## 🔥 Exception or Error

Error in browser console:
<pre><code>
Uncaught SyntaxError: Unexpected token n in JSON at position 499
</code></pre>

**Comment by SanderElias** [comment]
@kgajera This is because we had to close a possible (when using untrusted API's) security issue. I'll work on this bug today. we had another related issue too.

**Comment by SanderElias** [comment]
This is closed in Scully 1.0.11
